sway-fmt/src/code_builder_helpers.rs

```diff
@@ -192,9 +192,16 @@ pub fn get_new_line_pattern(line: &str) -> Option<&str> {
 pub fn get_already_formatted_line_pattern(line: &str) -> Option<(&str, &str)> {
     let pattern_len = ALREADY_FORMATTED_LINE_PATTERN.len();
 
-    if line.len() >= pattern_len && &line[0..pattern_len] == ALREADY_FORMATTED_LINE_PATTERN {
-        // already formatted single lines end with ';'
-        let end = line.find(';').unwrap();
+    if line.starts_with(ALREADY_FORMATTED_LINE_PATTERN) {
+        let char_idxs = vec![
+            line.find(';').unwrap_or(0),
+            line.rfind(',').unwrap_or(0),
+            line.rfind('}').unwrap_or(0),
+            line.rfind('{').unwrap_or(0),
+        ];
+
+        let end = char_idxs.iter().max().unwrap_or(&0);
```

Comment:
- This will panic if (for whatever reason) that `0` default is used. 
  - Not sure how we would prefer to handle this
- While testing I did not see a panic when using properly formatted input (only if someone accidentally has a typo or something)

Reply:
If it will panic on `0`, why not just `unwrap()`?